### PR TITLE
[54926] Improved token validation in reporting flow

### PR DESCRIPTION
**Pull request by jonasofrich**
Implemented validation of token in every step of the reporting flow, and redirection to error page if validation of token fails

### CODE REVIEW

**Comment by jonasofrich** [comment]
> @jonasofrich is it agreed to be included in 2.2?

Good question. @mmn-nc ?

**Comment by mmn-nc** [comment]
> > @jonasofrich is it agreed to be included in 2.2?
> 
> Good question. @mmn-nc ?

This is not a part of the 2.2 release. We might include it, but it is an edge-case.

File: NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/QuestionnaireViewController.cs [hunk 1]
```diff
@@ -170,6 +170,9 @@ void OnValidationFail()
         void OnFail()
         {
             NextBtn.HideSpinner();
+            AuthErrorUtils.GoToTechnicalError(this, LogSeverity.ERROR, null,
```
Comment (nc-kano) [inline NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/QuestionnaireViewController.cs:173]: and what with validation fail?

**Review by nc-kano — CHANGES_REQUESTED** [review]

File: NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/QuestionnaireViewController.cs (hunk 1 above)
Comment (jonasofrich) [reply NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/QuestionnaireViewController.cs:173]: It's only invoked if the sympton onset date is not selected, and in that case it displays a dialog.

**Review by nc-kano — APPROVED** [review]

**Pull request merged**